### PR TITLE
Adding webpack-dev-server

### PR DESCRIPTION
**Pull request by washingtonsoares**
Hey folks!

I really enjoyed this starter project and it helped me a lot when i was looking to start a project.

However I realized that it was possible to improve it in some points, among them:

- Using webpack-dev-server:
With webpack-dev-server it is possible to have a greater integration with webpack,
  Besides being a server with a very simple configuration to serve the webpack bundles.
 
- Using Gulp only to generate new components:
On Gulp I think it would only be necessary to generate new components, 
because the responsibles for generating the bundles would be webpack and the server for webpack-dev-server.

Also I have updated some dev dependencies, like the loaders and the webpack itself.

I believe that these changes will bring improvements and more users to this
 starter project and will certainly continue to help many devs who are starting a project!

Kindly look at this PR, and Come to me if you need anything to be adjusted.

Thanks!

### CODE REVIEW

**Comment by jayrmotta** [comment]
:+1:

**Comment by jvpsousa** [comment]
👍

File: package.json [hunk 1]
```diff
@@ -10,55 +10,45 @@
...
+    "yargs": "^6.6.0"
   },
   "scripts": {
-    "build": "gulp webpack",
-    "component": "gulp component",
-    "serve": "gulp serve",
-    "start": "gulp serve",
+    "build": "webpack --progress --config ./webpack.dist.config.js",
+    "serve": "webpack-dev-server --progress --config ./webpack.dev.config.js",
+    "start": "webpack-dev-server --progress --config ./webpack.dev.config.js",
```
Comment (fesor) [inline package.json]: maybe we should move all webpack configs to separate directory?

Comment (washingtonsoares) [inline package.json]: Good idea
I will refactor and update the PR.

**Comment by andrei-dracea** [comment]
Who else needs to review this in order to merge it? What are the next steps?

**Comment by fesor** [comment]
The only problem that I see with this PR is that it removes `browser-sync`. This is pretty useful tool for cross-device testing.

Also there was some stuff which could be improved in webpack configuration management.

**Review by epworthhh — APPROVED** [review]

**Comment by gdi2290** [comment]
@fesor is this PR good?

**Comment by scragg0x** [comment]
Could this plugin be used for browser-sync? I have never used it myself.
https://www.npmjs.com/package/browser-sync-webpack-plugin

**Comment by thematho** [comment]
I took this branch and it's brand new Jest enabled configuration It works like a charm, I merged into my improved plop generators branch with full examples of Testing AngularJS Components, Directives and Services with Jest:
[Improved plop generators, merged with Webpack 4, Webpack Dev server and Jest](https://github.com/thematho/NG6-starter/tree/webpack_update)

**Pull request closed without merging**